js/data.js

```diff
@@ -6,3 +6,16 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var previousJournalEntries = localStorage.getItem('javascript-local-storage');
+if (previousJournalEntries !== null) {
+  data.entries = JSON.parse(previousJournalEntries);
+  data.nextEntryId += data.entries.length;
+}
+
+window.addEventListener('beforeunload', beforeUnloadHandle);
+
+function beforeUnloadHandle(event) {
+  var dataJSON = JSON.stringify(data.entries);
```

Comment:
You should save the entire `data` variable into localStorage, not just the `entries` property. Make sure you clear out localStorage after this change so you don't end up with corrupted data 🤓